flusher.go

```diff
@@ -110,31 +177,44 @@ func (s *Server) Flush(interval time.Duration, metricLimit int) {
 	}
 
 	finalizeMetrics(s.Hostname, s.Tags, finalMetrics)
-
 	s.statsd.TimeInMilliseconds("flush.total_duration_ns", float64(time.Now().Sub(combineStart).Nanoseconds()), []string{"part:combine"}, 1.0)
 
-	s.statsd.Count("worker.metrics_flushed_total", int64(totalCounters), []string{"metric_type:counter"}, 1.0)
-	s.statsd.Count("worker.metrics_flushed_total", int64(totalGauges), []string{"metric_type:gauge"}, 1.0)
-	s.statsd.Count("worker.metrics_flushed_total", int64(totalLocalHistograms), []string{"metric_type:local_histogram"}, 1.0)
-	s.statsd.Count("worker.metrics_flushed_total", int64(totalLocalSets), []string{"metric_type:local_set"}, 1.0)
-	s.statsd.Count("worker.metrics_flushed_total", int64(totalLocalTimers), []string{"metric_type:local_timer"}, 1.0)
-	if s.ForwardAddr == "" {
-		// only report these lengths if we're the global veneur instance
-		// responsible for flushing them
-		// this avoids double-counting problems where a local veneur reports
-		// histograms that it received, and then a global veneur reports them
-		// again
-		s.statsd.Count("worker.metrics_flushed_total", int64(totalHistograms), []string{"metric_type:histogram"}, 1.0)
-		s.statsd.Count("worker.metrics_flushed_total", int64(totalSets), []string{"metric_type:set"}, 1.0)
-		s.statsd.Count("worker.metrics_flushed_total", int64(totalTimers), []string{"metric_type:timer"}, 1.0)
-	}
+	return finalMetrics
+}
 
-	if s.ForwardAddr != "" {
-		// we cannot do this until we're done using tempMetrics here, since
-		// not everything in tempMetrics is safe for sharing
-		go s.flushForward(tempMetrics)
-	}
+// reportMetricsFlushCounts reports the counts of
+// Counters, Gauges, LocalHistograms, LocalSets, and LocalTimers
+// to Datadog. These are shared by both global and local flush operations.
```

Comment:
Nit: Instead of "to Datadog" use "as metrics" since where these go isn't really Veneur's concern. Just out into the UDP ether! 😄